### PR TITLE
Replace github.com/pkg/errors with standard library errors package

### LINKED ISSUE #1678: Replace github.com/pkg/errors with standard library errors package
This is to use the errors package instead of `github.com/pkg/errors` as appropriate. Further, when base "errors" is imported it is better to rename the import of k8s.io/apimachinery/pkg/api/errors` as k8serrors across.

For example this would help deal with this [comment](https://github.com/RamenDR/ramen/pull/1674/commits/72abcab67f1931c309ea18c10d7d52adc8fc4e9f#r1855461254) better.

### PR DESCRIPTION
**Pull request by nirs**
- Import apimachinery api errors as k8serrors
- Replace github.com/pkg/errors with standard errors
- Fix errors to make revive happy

Fixes #1678

### CODE REVIEW

File: internal/controller/drplacementcontrol_controller.go [hunk 1]
```diff
@@ -60,7 +60,7 @@ const (
 	DoNotDeletePVCAnnotationVal = "true"
 )
 
-var InitialWaitTimeForDRPCPlacementRule = errorswrapper.New("Waiting for DRPC Placement to produces placement decision")
+var ErrInitialWaitTimeForDRPCPlacementRule = errors.New("waiting for DRPC Placement to produces placement decision")
```
Comment (BenamarMk) [inline internal/controller/drplacementcontrol_controller.go:63]: While you are at it, remve the `s` from `produces`

**Review by BenamarMk — APPROVED** [review]

File: internal/controller/drplacementcontrol_controller.go (hunk 1 above)
Comment (nirs) [reply internal/controller/drplacementcontrol_controller.go]: fixed

**Comment by nirs** [comment]
This fixes the issue only in one file, we need to fix the entire project, but it should be ready for merge.

**Review by ShyamsundarR — APPROVED** [review]

**Comment by nirs** [comment]
Fixed the conflicts, should be ready for merge when CI completes.

**Pull request merged**